### PR TITLE
III-4601 Add hash to place to make it unique

### PR DESCRIPTION
**Pull request by grubolsch**
### Added
Add hash to place to make it unique

---

Ticket: https://jira.uitdatabank.be/browse/III-4601

### CODE REVIEW

File: src/ElasticSearch/JsonDocument/Properties/AddUniqueHashToPlaceTransformer.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            $from['address'][$lang]['addressLocality'] ?? '',
+            $from['address'][$lang]['addressCountry'] ?? '',
+            $from['creator'] ?? '',
+        ];
+
+        //we trim both sides of each part, and remove the empty parts
+        $value = mb_strtolower(implode('_', array_filter(array_map('trim', $parts))));
+
+        if (!empty($value)) {
+            $draft['hash'] = $value;
```
Comment (LucWollants) [inline src/ElasticSearch/JsonDocument/Properties/AddUniqueHashToPlaceTransformer.php:28]: I would make this a bit more explicit. Something like `unique_address_identifier`

Comment (LucWollants) [reply src/ElasticSearch/JsonDocument/Properties/AddUniqueHashToPlaceTransformer.php:28]: If a more explicit name like that is used, it is also best to rename `AddUniqueHashToPlaceTransformer` to something like `UniqueAddressTransformer`

File: tests/ElasticSearch/JsonDocument/Properties/AddUniqueHashToPlaceTransformerTest.php [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                            'postalCode' => '2000',
+                            'addressLocality' => 'Antwerpen',
+                            'addressCountry' => 'BE',
+                        ],
+                    ],
+                    'creator' => 'John Doe',
+                ],
+                'dansstudio_teststraat 1_2000_antwerpen_be_john doe',
+            ],
+            'no man language' => [
```
Comment (LucWollants) [inline tests/ElasticSearch/JsonDocument/Properties/AddUniqueHashToPlaceTransformerTest.php:65]: Typo in `man`?

Comment (grubolsch) [reply tests/ElasticSearch/JsonDocument/Properties/AddUniqueHashToPlaceTransformerTest.php:65]: Good catch

**Review by LucWollants — APPROVED** [review]

**Pull request merged**